security/imx_hab4/sign-file.sh

```diff
@@ -140,45 +140,45 @@ sed "s^@@KEY_ROOT@@^${KEY_DIR}^g" ${CSF_TEMPLATE} > ${CSF_TEMPLATE}.csf-config
 IMG=$(ls -t ${KEY_DIR}/IMG${SRK_INDEX}*_crt.pem 2> /dev/null | head -1)
 if [ -n "${IMG}" ]
 then
-    sed -i "s^${KEY_DIR}/IMG_1_crt.pem^${IMG}^" ${CSF_TEMPLATE}.csf-config
+    sed -i~ "s^${KEY_DIR}/IMG_1_crt.pem^${IMG}^" ${CSF_TEMPLATE}.csf-config
```

Comment:
What does the tilde do here?  I've never seen that before.

Reply:
the ~ is identified as the suffix of a backup file. on linux if it is missing no backup file is generated, on mac there must be an argument. If you make the argument blank on mac it works (-i "") but that does not work on linux so I specified a backup with a ~ suffix and it works on both linux and mac.